### PR TITLE
exp/lighthorizon, xdr: Rename `CheckpointIndex` to better reflect its capabilty.

### PR DESCRIPTION
**Pull request by Shaptic**
### What
Rename `{types,xdr}.CheckpointIndex` -> `{types,xdr}.BitmapIndex`.  
Rename `NextActive` -> `NextActiveBit`.

### Why
This is a better reflection of the capability of the index: it can operate on any value, not necessarily just checkpoints (this is further evidenced by using it to index by ledger).

### Known limitations
n/a

### CODE REVIEW

File: exp/lighthorizon/index/types/bitmap.go [hunk 1]
```diff
@@ -29,83 +29,83 @@ func NewCheckpointIndex(b []byte) (*CheckpointIndex, error) {
 	return NewCheckpointIndexFromXDR(xdrCheckpoint), nil
 }
 
-func NewCheckpointIndexFromXDR(index xdr.CheckpointIndex) *CheckpointIndex {
-	return &CheckpointIndex{
-		bitmap:          index.Bitmap[:],
-		firstCheckpoint: uint32(index.FirstCheckpoint),
-		lastCheckpoint:  uint32(index.LastCheckpoint),
+func NewCheckpointIndexFromXDR(index xdr.BitmapIndex) *BitmapIndex {
```
Comment (sreuland) [inline exp/lighthorizon/index/types/bitmap.go:32]: to be consistent on naming, would this be better as `NewBitmapIndexFromXDR` ?

File: exp/lighthorizon/index/types/bitmap.go [hunk 2]
```diff
@@ -151,55 +151,55 @@ func (i *CheckpointIndex) iterate(f func(checkpoint uint32)) error {
...
 		}
-		err = i.setActive(checkpoint)
+		err = i.setActive(index)
 	})
 
 	return err
 }
 
-// NextActive returns the next checkpoint (inclusive) where this index is
+// NextActiveBit returns the next checkpoint (inclusive) where this index is
```
Comment (sreuland) [inline exp/lighthorizon/index/types/bitmap.go:169]: adjust docs a bit per rename, like:
```
Returns the next active index location (inclusive). 
If index is active at 'index' then it returns same 'index'.
```

**Review by sreuland — APPROVED** [review]
looks good, left minor suggestions.

File: exp/lighthorizon/index/types/bitmap.go (hunk 1 above)
Comment (Shaptic) [reply exp/lighthorizon/index/types/bitmap.go]: good catch! ctrl+f isn't perfect, heh. done in cf37fe77.

File: exp/lighthorizon/index/types/bitmap.go (hunk 2 above)
Comment (Shaptic) [reply exp/lighthorizon/index/types/bitmap.go]: yep, great call! done in cf37fe77.

**Pull request merged**